### PR TITLE
Panel fix

### PR DESCRIPTION
**Pull request by leocd91**
added:

- file selection, automatically list .npy available on ./image_and_data/, added a load button because the select's @pn.depends("select_file.value") don't work.
- reset button, version 1 (zero'd the filter)
- requirements.txt

fix:

- save button, added output.seek(0) so the downloaded file won't be empty.

tweaks:

- difference plot's `clim`
- readme on how to run it

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,30 @@ To deactivate the environment, use
      $ conda deactivate
 ```
 
+## Requirements
```
Comment (mycarta) [inline README.md:32]: This looks good

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+scipy==1.5.0
+scikit_image==0.16.2
+holoviews==1.14.3
+torch==1.8.1
+dask==2.19.0
+panel==0.11.3
+hvplot==0.7.1
+xarray==0.17.0
+numpy==1.20.2
+skimage==0.0
```
Comment (mycarta) [inline requirements.txt:13]: Good

File: smallfoot/app.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    listfile = []
+    listname = []
+    for root, dirs, files in os.walk(r'./images_and_data/'):
+        for file in files:
+            file2=file.lower()
+            if file2.endswith(".sgy") or file2.endswith(".npy"):
+               listfile.append(os.path.join(root, file))
+               listname.append(file)
+    fd = FilterDesigner(listfile)
+    fd.app().servable("Filter Designer")
```
Comment (mycarta) [inline smallfoot/app.py:258]: I am going to have to merge and run the app to really be able to test.

**Review by mycarta — COMMENTED** [review]
@leocd91 this review thing is new to both of us as per our slack chat.
By just looking at the changes I do not see any issues, I will jsut have to merge, then run the app to give it a go.

@WesleyTheGeolien any suggestions? Are you familiar with how this works?

File: smallfoot/app.py (hunk 3 above)
Comment (leocd91) [reply smallfoot/app.py:258]: I think you can test it by cloning this branch first @mycarta :
`git clone -b panel_fix https://github.com/mycarta/t21-hack-footprint/`

**Comment by WesleyTheGeolien** [comment]
> @leocd91 this review thing is new to both of us as per our slack chat.
> By just looking at the changes I do not see any issues, I will jsut have to merge, then run the app to give it a go.
> 
> @WesleyTheGeolien any suggestions? Are you familiar with how this works?

Yep we can check code implementations and such here sorry I havent had too much time to look over this but I was hoping to check the apply button implementations to see if we could get it working automatically as I believe it is not?

File: smallfoot/app.py (hunk 3 above)
Comment (mycarta) [reply smallfoot/app.py:258]: I will try it this weekend @leocd91

Comment (mycarta) [reply smallfoot/app.py:258]: Cool, you cna do this in GitHUb desktop too!

Screen Shot 2021-05-10 at 12 07 55 PM

Comment (leocd91) [reply smallfoot/app.py:258]: wait there is a github desktop? 😆

Comment (mycarta) [reply smallfoot/app.py:258]: Yup, I really like it

**Pull request merged**